modules/ROOT/pages/feature-flagging.adoc

```diff
@@ -0,0 +1,162 @@
+
+By default, Mule runtime engine disables any new features or bug fixes that are not backward compatible with Mule applications running in earlier Mule versions. However, it is possible to manage the feature flags configured for each Mule application or all apps running in the same Mule instance.
+
+== Configure Feature Flags for a Mule Application
+
+To configure feature flags for a Mule application, change the `minMuleVersion` value in the `mule-artifact.json` file of your app. This configuration instructs the Mule instance to run this application implementing all features and bug fixes that were released to that particular version and disabling features and bug fixes that are active by default in later Mule versions.
+//Think of adapting this to versionless.
+
+For example, you can run a Mule app in Mule 4.4.0 instructing the Mule instance to apply only the features that do not change core functionality up to Mule 4.1.0:
+
+.Example `mule-artifact.json` file
+----
+{
+  "minMuleVersion": "4.1.0"
+}
+----
+
+== Configure Feature Flags for the Mule Instance
+
+An alternative way to configure feature flags is by using system properties. In this case, you can enable or disable each one of the available feature flags. However, configuring feature flags through system properties applies the changes at the Mule instance level, so all applications deployed to this Mule instance behave according to the configured feature flags.
+
+Note that this configuration is compatible with on-premise deployments only because you cannot configure system properties in other deployment targets.
+
+For example, you can configure a Mule 4.3.0 instance to enable and disable specific features for all applications running on this instance:
+
+.Example `wrapper.conf` file
+----
+…
+wrapper.java.additional.999=-DHONOUR_RESERVED_PROPERTIES_PROPERTY=true
+wrapper.java.additional.1000=-DCOMPUTE_CONNECTION_ERRORS_IN_STATS_PROPERTY=false
+----
+
+== Feature Flags Reference
+
+//Confirm this statement which is not clear yet.
+The following table shows the available feature flags, a description of their functionality, the Mule version in which the flag became available, the earlier minor version releases that enable the flag by default, and the issue ID where the change was implemented:
+
+[%header%autowidth.stretch,cols="a,a,a"]
+|===
+|Feature Flag | Description | Issue ID
+<.^|`HONOUR_RESERVED_PROPERTIES_PROPERTY`
+|When enabled, reserved properties such as `app.name` can't be overridden by global properties (overrides are ignored).
+
+*Available Since*
+
+* 4.4.0
+* 4.3.0-202102
+
+*Enabled by Default Since*
+
+* 4.4.0
+* 4.3.1
+<.^| `MULE-17659`
+<.^|`HANDLE_SPLITTER_EXCEPTION_PROPERTY`
+|When enabled, DataWeave handles splitter exceptions and prevents Scala serialization issues.
```

Comment:
not sure if customers are going to get this one (splitter exceptions, Scala)

Reply:
Checking with the DW team. Will update with conclusions asap.

Reply:
Updated in commit af02ddc, please confirm if it is ok.